test-js/unit/pentaho/util/error.Spec.js

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright 2010 - 2015 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define(["pentaho/util/error"], function(error) {
+
+  /*global describe:false, it:false, expect:false, beforeEach:false, afterEach:false, spyOn:false, JSON:false*/
+
+  describe("pentaho.util.error -", function() {
+    it("is defined", function() {
+      expect(error instanceof Object).toBe(true);
+    });
+
+    function itShouldHaveMessage(methodName, withArgs, message) {
+      it("should have message «" + message + "» when called with arguments " + JSON.stringify(withArgs), function() {
+
+        var result = error[methodName].apply(error, withArgs);
+
+        expect(result instanceof Error).toBe(true);
+        expect(result.message).toBe(message);
+      });
+    }
+
+    describe("argRequired(name, (text)) -", function() {
+      it("should be a function", function() {
+        expect(typeof error.argRequired).toBe("function");
+      });
+
+      itShouldHaveMessage("argRequired", ["foo"       ], "Argument required: 'foo'.");
+      itShouldHaveMessage("argRequired", ["foo", "bar"], "Argument required: 'foo'. bar.");
+    });
+
+    describe("argInvalid(name, (reason)) -", function() {
+      it("should be a function", function() {
+        expect(typeof error.argInvalid).toBe("function");
+      });
+
+      itShouldHaveMessage("argInvalid", ["foo"       ], "Argument invalid: 'foo'.");
+      itShouldHaveMessage("argInvalid", ["foo", "bar"], "Argument invalid: 'foo'. bar.");
+    });
+
+    describe("argInvalidType(name, [expectedType], (gotType)) -", function() {
+      it("should be a function", function() {
+        expect(typeof error.argInvalidType).toBe("function");
+      });
+
+      itShouldHaveMessage("argInvalidType", ["foo", "string"],
+          "Argument invalid: 'foo'. Expected type to be string.");
```

Comment:
Ok, now I fully understand your arguments on #620 concerning thrown errors. If you are testing the messages carried by the thrown errors here, I'm more comfortable with having other tests depend on this module.